src/backend/drm/mod.rs

```diff
@@ -462,61 +517,99 @@ pub trait DrmHandler<B: Borrow<DrmBackend> + 'static> {
     ///
     /// The `id` argument is the `Id` of the `DrmBackend` that finished rendering,
     /// check using `DrmBackend::is`.
-    fn ready(&mut self, evlh: &mut EventLoopHandle, device: &mut DrmDevice<B>, backend: &StateToken<B>,
-             frame: u32, duration: Duration);
+    fn ready<'a, S: Into<StateProxy<'a>>>(&mut self, state: S, device: &mut DrmDevice<B>, backend: &StateToken<B>,
```

Comment:
In general for the methods providing an `Into<StateProxy>` rather than the full `EventLoopHandle`, it would be good to document if some state keys known to the user are susceptible to be already borrowed (and thus trying to borrow them again would result in a panic).

If the only already-borrowed state keys are implementation details (not known to smithay's user), no need to obviously.